google-cloud-core/src/main/java/com/google/cloud/HttpTransportOptions.java

```diff
@@ -138,7 +138,8 @@ public HttpTransportFactory getHttpTransportFactory() {
    * Returns a request initializer responsible for initializing requests according to service
    * options.
    */
-  public HttpRequestInitializer getHttpRequestInitializer(ServiceOptions<?, ?> serviceOptions) {
+  public HttpRequestInitializer getHttpRequestInitializer(
+      ServiceOptions<?, ?> serviceOptions, final String artifactId) {
```

Comment:
I think everything would be a lot easier if the artifact id was put into `ServiceOptions` itself (passed into the constructor from the service-specific derived options class).